src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -1214,6 +1217,15 @@ private static CompletionItemBuilder CreateModulePathCompletionBuilder(string na
                 .WithSortText(GetSortText(label, priority))
                 .Build();
 
+        //we need to check for overflow due to using code blocks.
+        //if we reach limit in a code block vscode will truncate it automatically, the block will not be terminated so the hover will not be properly formatted
+        //therefore we need to check for the limit ourselves and truncate text inside code block, making sure it's terminated properly.
+        private static string CodeBlock(string content) =>
+        $"```bicep\n{(content.Length > MaxHoverMarkdownCodeBlockLength ? content.Substring(0, MaxHoverMarkdownCodeBlockLength) : content)}\n```\n";
+        
+        // Markdown needs two leading whitespaces before newline to insert a line break
+        private static string CodeBlockWithDescription(string content, string? description) => CodeBlock(content) + (description is not null ? $"{description.Replace("\n", "  \n")}\n" : string.Empty);
```

Comment:
>description is not null ? $"{description.Replace("\n", "  \n")}\n" : string.Empty [](http://example.com/codeflow?start=117&length=81)

The description formatting logic should probably be factored out into its own method.